### PR TITLE
add include and exclude options to readSync

### LINKED ISSUE #8: includeFiles options
Would you support a `includeFiles` option for readSync? I know what files subset I want, but I like the object nesting I get from this library.

### PR DESCRIPTION
**Pull request by kellyselden**
Closes #8.

This will really help out https://github.com/kellyselden/ember-cli-update/pull/108.

cc @rwjblue.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,11 +1,34 @@
...
+  var includeFiles = options.includeFiles
+  var excludeFiles = options.excludeFiles || []
+
   var obj = {}
   var entries = fs.readdirSync(dir).sort()
   for (var i = 0; i < entries.length; i++) {
     var entry = entries[i]
+
+    if (includeFiles) {
+      if (!performMatch(entry, includeFiles)) {
```
Comment (rwjblue) [inline index.js]: Entry here is just going to be the filename, not the path portion. I would expect to be able to match things based on the full path (not just the filename)...

**Review by rwjblue — COMMENTED** [review]
👍 - The idea seems good to me.  Should we use `include` and `exclude` instead of `{include,exclude}Files`? broccoli-funnel uses `include`/`exclude`, and it doesn't seem (based on the changes made here) that the patterns are _only_ used against files anyways... 🤔

File: index.js (hunk 1 above)
Comment (kellyselden) [reply index.js]: Good catch

**Comment by kellyselden** [comment]
Updated names

File: index.js (hunk 1 above)
Comment (kellyselden) [reply index.js]: Fixed the bug, and altered the tests to catch it

**Comment by kellyselden** [comment]
Should be good to go now.

File: index.js [hunk 2]
```diff
@@ -1,17 +1,43 @@
+var path = require('path')
 var fs = require('fs-extra')
+var minimatch = require('minimatch')
+
+function performMatch (entry, globs) {
```
Comment (stefanpenner) [inline index.js]: should we use: https://www.npmjs.com/package/matcher-collection for this? As it is has tests: https://github.com/stefanpenner/matcher-collection/blob/master/test.js

Comment (kellyselden) [reply index.js]: I think so

Comment (stefanpenner) [reply index.js]: sounds good. I'll good with it either way. Your call

**Review by stefanpenner — APPROVED** [review]

File: index.js (hunk 2 above)
Comment (kellyselden) [reply index.js]: updated to use matcher-collection

File: index.js [hunk 3]
```diff
@@ -1,17 +1,41 @@
...
+  options = options || {}
+  var include = options.include
+  var exclude = options.exclude || []
+  relativeRoot = relativeRoot || '';
+
+  var includeMatcher
+  if (include) {
+    includeMatcher = new MatcherCollection(include)
+  }
+  var excludeMatcher = new MatcherCollection(exclude)
```
Comment (stefanpenner) [inline index.js]: if there is no exclude, should we bother with an excludeMatcher

Comment (kellyselden) [reply index.js]: changed to include that optimization

**Review by stefanpenner — APPROVED** [review]
LGTM - thanks

**Comment by stefanpenner** [comment]
released as v0.3.4 🎉

**Comment by kellyselden** [comment]
Woohoo Thanks!

**Pull request merged**